addons/mail/static/src/core/common/link_preview.xml

```diff
@@ -52,43 +52,52 @@
+                        <p t-if="props.linkPreview.og_description" class="o-mail-LinkPreviewVideo-description o-mail-LinkPreviewVideo-hasDescription card-text mb-0 text-muted small overflow-hidden" t-out="props.linkPreview.og_description"></p>
                     </div>
                 </div>
-                <div class="o-mail-LinkPreviewVideo-videoWrap col-4 d-block">
-                    <a t-att-href="props.linkPreview.source_url" target="_blank">
-                        <div t-if="props.linkPreview.og_image" class="o-mail-LinkPreviewVideo-overlay position-relative h-100 opacity-trigger-hover">
-                            <img class="img-fluid h-100 rounded-bottom o_object_fit_cover" t-att-src="props.linkPreview.og_image" t-att-alt="props.linkPreview.og_title" t-on-load="onImageLoaded"/>
-                            <div class="position-absolute top-50 start-50 translate-middle">
-                                <div class="o-mail-LinkPreviewVideo-play btn btn-lg rounded opacity-75 opacity-100-hover transition-base">
-                                    <i class="fa fa-play"/>
-                                </div>
-                            </div>
-                        </div>
-                        <span t-else="" class="d-flex align-items-center justify-content-center w-100 h-100 bg-100 text-300">
-                            <i class="fa fa-camera fa-2x"/>
-                        </span>
-                    </a>
+                <div t-if="!props.linkPreview.videoURL" class="o-mail-LinkPreviewVideo-videoWrap col-4 d-block">
+                    <a t-att-href="props.linkPreview.source_url" target="_blank"><t t-call="mail.LinkPreview.videoPreview"/></a>
                 </div>
             </div>
+            <div t-if="props.linkPreview.videoURL" class="o-linkPreviewVideo-container">
+                <div t-if="!state.videoLoaded" class="o-mail-LinkPreviewVideo-videoWrap h-100 w-100" t-on-click="() => state.startVideo = true"><t t-call="mail.LinkPreview.videoPreview"/></div>
+                <iframe t-if="state.startVideo" t-att-src="props.linkPreview.videoURL" class="h-100 w-100" t-att-class="{'d-none': !state.videoLoaded}" allow="autoplay" frameborder="0" allowFullScreen="true" t-ref="video"/>
+            </div>
             <t t-if="props.deletable" t-call="mail.LinkPreview.aside">
                 <t t-set="className" t-value="'fa fa-stack p-0 opacity-75 opacity-100-hover'"/>
             </t>
         </div>
     </t>
-    
+
     <t t-name="mail.LinkPreview.aside">
         <div class="position-absolute top-0 end-0 small">
             <button t-attf-class="{{ className }}" class="o-mail-LinkPreview-aside btn" aria-label="Remove" t-on-click="onClick">
                 <i class="fa fa-times"/>
             </button>
         </div>
     </t>
+
+    <t t-name="mail.LinkPreview.videoPreview">
```

Comment:
Nitpick: would name it `mail.LinkPreviewVideo.preview`. We split the different types of preview as if they were different components (before 16.3 they were actually different components). I would try to be consistent with current naming convention, prefer doing cleaning at once on whole template in a master PR (they this PR will be backported 18.0)